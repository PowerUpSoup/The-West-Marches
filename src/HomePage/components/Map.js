import React, { Component } from 'react';
import dmMap from './DM-Campaign-Map.jpg';
import playerMap from './Campaign Map.png';
import ApiContext from '../../ApiContext';

class Map extends Component {

    static contextType = ApiContext;

    render() {
        let sessionStorageUser = JSON.parse(sessionStorage.getItem("user"));
        if (sessionStorageUser.role === "dungeon_master") {
            return (
                <div>
                    <img className="Map" src={dmMap} alt="The Dungeon Master's map of Barovia" />
                </div>
            );
        }
        else if (sessionStorageUser.role === "player") {
            return (
                <div>
                    <img className="Map" src={playerMap} alt="The Player's map of Barovia" />
                </div>
            );
        } else {
            this.props.history.push('/');
            return (null);
        };
    };
};


export default Map;