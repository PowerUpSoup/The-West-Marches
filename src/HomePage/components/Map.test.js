  
import React from 'react';
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json'
import Map from './Map';

describe(`Map component`, () => {

    it('renders a Map', () => {
        const wrapper = shallow(
            <BrowserRouter>
                <Map />
            </BrowserRouter>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})