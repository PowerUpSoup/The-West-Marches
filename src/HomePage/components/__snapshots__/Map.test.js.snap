// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component renders a Map 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Map />
</Router>
`;
